drivers/input/touchscreen/atmel_mxt_ts.c

```diff
@@ -962,11 +1103,125 @@ static int __mxt_write_reg(struct i2c_client *client, u16 reg, u16 len,
 	return ret;
 }
 
+static int __mxt_write_reg_crc(struct i2c_client *client, u16 reg, u16 length,
+			   const void *val, struct mxt_data *data)
+{
+	u8 msgbuf[15];
```

Comment:
Please leave comment why it's 15 bytes. This is the limitation of maXTouch firmware. The 15 bytes are the max number of bytes allowed in a write transfer, including register address (2 bytes), seq. number (1 byte) and CRC8 (1 byte).

Reply:
For me, I have no concern, after all, as a driver developer, we're crafting an adapter between the firmware and the OS. One day if any new comer comes and reads this code, they may ask: "Why it's 15 here?" It's perfectly ok for me to state the "characteristic/limitation" of the firmware in the code to make the driver developers fully understand the code. The better they understand the firmware, the better code they can craft.

If you have concern, please ignore my comment. You lead the way.